public/components/datasources/components/new/configure_datasource.tsx

```diff
@@ -0,0 +1,211 @@
+  const ConfigureDatasourceSteps = [
+    {
+      title: 'Step 1',
+      children: (
+        <EuiText>
+          <p>Configure Connection</p>
+        </EuiText>
+      ),
+    },
+    {
+      title: 'Step 2',
+      children: (
+        <EuiText>
+          <p>Review Configuration</p>
+        </EuiText>
+      ),
+    },
+  ];
+
+  useEffect(() => {
+    http!.get('/api/v1/configuration/roles').then((data) =>
+      setRoles(
+        Object.keys(data.data).map((key) => {
+          return { label: key };
+        })
+      )
+    );
+  }, []);
+
+  const ConfigureDatasource = (configurationProps: { datasourceType: string }) => {
+    const { datasourceType } = configurationProps;
+    switch (datasourceType) {
+      case 'S3':
+        return (
+          <ConfigureS3Datasource
+            currentName={name}
+            currentDetails={details}
+            setNameForRequest={setName}
+            setDetailsForRequest={setDetails}
+            currentArn={arn}
+            setArnForRequest={setArn}
+            currentStore={store}
+            setStoreForRequest={setStore}
+            roles={roles}
+            selectedQueryPermissionRoles={selectedQueryPermissionRoles}
+            setSelectedQueryPermissionRoles={setSelectedQueryPermissionRoles}
+          />
+        );
+      default:
+        return <></>;
+    }
+  };
+
+  const ReviewDatasourceConfiguration = (configurationProps: { datasourceType: string }) => {
+    const { datasourceType } = configurationProps;
+    switch (datasourceType) {
+      case 'S3':
+        return (
+          <ReviewS3Datasource
+            currentName={name}
+            currentDetails={details}
+            currentArn={arn}
+            currentStore={store}
+            selectedQueryPermissionRoles={selectedQueryPermissionRoles}
+          />
+        );
+      default:
+        return <></>;
+    }
+  };
+
+  const ReviewSaveOrCancel = useCallback(() => {
+    return (
+      <EuiBottomBar>
+        <EuiFlexGroup justifyContent="flexEnd">
+          <EuiFlexItem grow={false}>
+            <EuiButtonEmpty
+              onClick={() => {
+                window.location.hash = '#/new';
+              }}
+              color="ghost"
+              size="s"
+              iconType="cross"
+            >
+              Cancel
+            </EuiButtonEmpty>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButton
+              onClick={page === 'review' ? () => setPage('configure') : () => {}}
+              color="ghost"
+              size="s"
+              iconType="arrowLeft"
+            >
+              Previous
+            </EuiButton>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiButton
+              onClick={
+                page === 'review'
+                  ? () => createDatasource()
+                  : () => {
+                      setPage('review');
+                    }
```

Comment:
any reason to not do the check inside the function? like `() => page === 'review' ? createDatasource() : setPage('review')`

Reply:
Not really - this is more clear, so changed to this in: [680763b](https://github.com/opensearch-project/dashboards-observability/pull/1049/commits/680763b566a24a70d93f79214331182f495ed291)